src/main/java/screens/FormplayerQueryScreen.java

```diff
@@ -27,14 +29,19 @@ public FormplayerQueryScreen(HqAuth auth){
         this.auth = auth;
     }
 
-    public String getUriString() {
+    public String getDecodedUriString() {
         URL url = getBaseUrl();
         Hashtable<String, String> params = getQueryParams();
         UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(url.toString());
         for(String key: params.keySet()){
             builder.queryParam(key, params.get(key));
         }
-        return builder.toUriString();
+        // In order to get a decoded URI we decode after building the URI.
+        try {
+            return URLDecoder.decode(builder.toUriString(), "UTF-8");
```

Comment:
Hmm no other thoughts. I'm pro just reverting back to your original fix, don't let the perfect be the enemy of the good and all